File: 백준/Python/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
n = int(input())

cnt = 0
for i in range(n):
    check = [0] * 26
    s = input()
    
    rp = -1
    result = True
    for ss in s:
        idx = ord(ss) - ord('a')
        if check[idx] != 0 and rp != idx:
            result = False
            break
        check[idx] += 1
        rp = idx
    if result:
        cnt += 1
        
print(cnt)

File: 백준/Java/Silver/1316. 그룹 단어 체커/그룹 단어 체커.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int cnt = 0;
        for(int i=0; i<n; i++) {
            String s = sc.next();
            boolean[] check = new boolean[26];
            boolean result = true;
            for(int j=0; j<s.length(); j++) {
                if(j>0 && s.charAt(j) != s.charAt(j-1) && check[s.charAt(j)-'a'] == true) {
                        result = false;
                        break;
                }
                check[s.charAt(j) - 'a'] = true;
            }
            if (result) {
                cnt++;
            }
     
        }
        
        System.out.print(cnt);
    }
}